.eslintrc.js

```diff
@@ -9,9 +8,14 @@ module.exports = {
     '@metamask/eslint-config/config/typescript',
```

Comment:
I would suggest putting the typescript rules in an override. This config disables a variety of ESLint rules that we still want to apply to JavaScript modules. Also the TypeScript lint rules can be incorrectly triggered by lint errors in JavaScript modules in some cases (I recently encountered this here: https://github.com/MetaMask/controllers/pull/396